lib/list_versions.go

```diff
@@ -183,34 +190,35 @@ func removeDuplicateVersions(elements []string) []string {
 	return result
 }
 
-// validVersionFormat : returns valid version format
-/* For example: 0.1.2 = valid
-// For example: 0.1.2-beta1 = valid
-// For example: 0.1.2-alpha = valid
-// For example: a.1.2 = invalid
-// For example: 0.1. 2 = invalid
-*/
-func validVersionFormat(version string) bool {
-	// Getting versions from body; should return match /X.X.X-@/ where X is a number,@ is a word character between a-z or A-Z
-	// Follow https://semver.org/spec/v1.0.0-beta.html
-	// Check regular expression at https://rubular.com/r/ju3PxbaSBALpJB
-	semverRegex := regexp.MustCompile(`^(\d+\.\d+\.\d+)(-[a-zA-z]+\d*)?$`)
-	return semverRegex.MatchString(version)
-}
+// validVersionFormat : Return true if valid semantic version provided based on the type of version requested
+func validVersionFormat(version string, reqType ...string) bool {
+	var verType string
+	var semverRegex *regexp.Regexp
+
+	switch len(reqType) {
+	case 0:
+		verType = "full"
+	case 1:
+		verType = reqType[0]
+	default:
+		logger.Fatalf("Internal error: invalid number of arguments (must be 2, got %d)", 1+len(reqType))
+	}
 
-// ValidMinorVersionFormat : returns valid MINOR version format
-/* For example: 0.1 = valid
-// For example: a.1.2 = invalid
-// For example: 0.1.2 = invalid
-*/
-func validMinorVersionFormat(version string) bool {
-	// Getting versions from body; should return match /X.X./ where X is a number
-	semverRegex := regexp.MustCompile(`^(\d+\.\d+)$`)
+	switch verType {
```

Comment:
As suggested, this seems quite overly verbose and relying on magic strings (and duplicates the names of the keys in the map).

Reply:
Won't be able to get rid of magic strings at this moment as regexps are used with different anchors throughout this file. The rest of comments and ideas from the other thread are being worked on towards implementation.